homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -1,291 +1,371 @@
 
     @property
-    def volume_level(self):
-        """Volume level of the media player (0..1)."""
-        return self.volume
+    def is_volume_muted(self):
+        """Return boolean if volume is currently muted."""
+        return self.coordinator.data.zones[self._zone_id].mute
 
     @property
-    def supported_features(self):
-        """Flag of features that are supported."""
-        return SUPPORTED_FEATURES
+    def shuffle(self):
+        """Boolean if shuffling is enabled."""
+        return (
+            self.coordinator.data.netusb_shuffle == "on" if self._is_netusb else False
+        )
 
     @property
-    def source(self):
-        """Return the current input source."""
-        return self._source
+    def sound_mode(self):
+        """Return the current sound mode."""
+        return self.coordinator.data.zones[self._zone_id].sound_program
 
     @property
-    def source_list(self):
-        """List of available input sources."""
-        return self._source_list
+    def sound_mode_list(self):
+        """Return a list of available sound modes."""
+        return self.coordinator.data.zones[self._zone_id].sound_program_list
 
-    @source_list.setter
-    def source_list(self, value):
-        """Set source_list attribute."""
-        self._source_list = value
+    @property
+    def device_class(self):
+        """Return the device class of the media player."""
+        return self._device_class
 
     @property
-    def media_content_type(self):
-        """Return the media content type."""
-        return MEDIA_TYPE_MUSIC
+    def zone(self):
+        """Return the zone of the media player."""
+        return self._zone_id
 
     @property
-    def media_duration(self):
-        """Duration of current playing media in seconds."""
-        return self.media_status.media_duration if self.media_status else None
+    def unique_id(self) -> str:
+        """Return the unique ID for this media_player."""
+        macs = self.coordinator.data.mac_addresses
+        return f"{macs}_{self._zone_id}"
+
+    async def async_turn_on(self):
+        """Turn the media player on."""
+        await self.coordinator.musiccast.turn_on(self._zone_id)
+        self.schedule_update_ha_state()
+
+    async def async_turn_off(self):
+        """Turn the media player off."""
+        await self.coordinator.musiccast.turn_off(self._zone_id)
+        self.schedule_update_ha_state()
+
+    async def async_mute_volume(self, mute):
+        """Mute the volume."""
+
+        await self.coordinator.musiccast.mute_volume(self._zone_id, mute)
+        self.schedule_update_ha_state()
+
+    async def async_set_volume_level(self, volume):
+        """Set the volume level, range 0..1."""
+        await self.coordinator.musiccast.set_volume_level(self._zone_id, volume)
+        self.schedule_update_ha_state()
+
+    async def async_media_play(self):
+        """Send play command."""
+        if self._is_netusb:
```

Comment:
Should we somehow handle the case if a user calls this service without it being supported, eg by raising an error?

Reply:
This will happen if the source is a live source like tuner (radio) or a line in, where play wouldn't do anything. I think its not really an error, but I'm interested in your opinion

Reply:
Ok. But then the user may be confused why nothing happens when this service is called in that case.

Reply:
Normally we raise an exception in the service handler if a service is incorrectly called. We could raise eg `HomeAssistantError("Service media play is not supported without netusb")`.

Reply:
Bump.

Reply:
I have mixed feelings about this, but I will investigate and check, how other integrations for AV-Receivers handle this - they should have the same problem and maybe we find a way to have this consistent